src/openrct2/object/ObjectTypes.h

```diff
@@ -66,17 +66,11 @@ constexpr std::array ObjectTypes = {
 // Not using EnumValue to reduce header includes
 static_assert(ObjectTypes.size() == static_cast<uint8_t>(ObjectType::Count));
 
-// Object types that can be saved in a park file.
-constexpr std::array<ObjectType, 16> TransientObjectTypes = {
-    ObjectType::Ride,         ObjectType::SmallScenery, ObjectType::LargeScenery,    ObjectType::Walls,
-    ObjectType::Banners,      ObjectType::Paths,        ObjectType::PathAdditions,   ObjectType::SceneryGroup,
-    ObjectType::ParkEntrance, ObjectType::Water,        ObjectType::TerrainSurface,  ObjectType::TerrainEdge,
-    ObjectType::Station,      ObjectType::Music,        ObjectType::FootpathSurface, ObjectType::FootpathRailings,
-};
-
-// Object types that cannot be saved in a park file.
-constexpr std::array<ObjectType, 2> IntransientObjectTypes = { ObjectType::ScenarioText, ObjectType::Audio };
+static constexpr size_t kNumTransientObjectTypes = 16;
+static constexpr size_t kNumIntransientObjectTypes = 2;
+static_assert(kNumTransientObjectTypes + kNumIntransientObjectTypes == static_cast<size_t>(ObjectType::Count));
```

Comment:
Move this into the cpp file, don't need to have this check for each compilation unit.